ocaml/lambda/lambda.ml

```diff
@@ -491,6 +497,14 @@ let make_key e =
         let y = make_key x in
         Llet (str,k,y,ex,tr_rec (Ident.add x (Lvar y) env) e)
     | Lprim (p,es,_) ->
+        let p =
+          (* The field read semantics is ignored when making the key but
+             joined when constructing a shared action (see below). *)
+          match p with
+          | Pfield (index, _sem) ->
+            Pfield (index, Reads_agree (* arbitrary *) )
```

Comment:
Whilst I agree it is basically arbitrary, I think that using `Reads_vary` is slightly better because it means that the key is actually a safe replacement for any of the actions that generate it.